java/src/main/native/ai_onnxruntime_OrtSession.c

```diff
@@ -13,227 +13,286 @@
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    size_t numInputs;
-    checkOrtStatus(jniEnv,api,api->SessionGetInputCount((OrtSession*)handle, &numInputs));
-    return numInputs;
+JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OrtSession_getNumInputs(JNIEnv* jniEnv, jobject jobj, jlong apiHandle, jlong handle) {
+  (void)jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  size_t numInputs = 0;
+  checkOrtStatus(jniEnv, api, api->SessionGetInputCount((OrtSession*)handle, &numInputs));
+  return numInputs;
 }
 
 /*
  * Class:     ai_onnxruntime_OrtSession
  * Method:    getInputNames
  * Signature: (JJJ)[Ljava/lang/String;
  */
-JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_getInputNames
-  (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong sessionHandle, jlong allocatorHandle) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    OrtAllocator* allocator = (OrtAllocator*) allocatorHandle;
-
-    // Setup
-    char *stringClassName = "java/lang/String";
-    jclass stringClazz = (*jniEnv)->FindClass(jniEnv, stringClassName);
-
-    // Get the number of inputs
-    size_t numInputs = Java_ai_onnxruntime_OrtSession_getNumInputs(jniEnv, jobj, apiHandle, sessionHandle);
-
-    // Allocate the return array
-    jobjectArray array = (*jniEnv)->NewObjectArray(jniEnv,safecast_size_t_to_jsize(numInputs),stringClazz,NULL);
-    for (uint32_t i = 0; i < numInputs; i++) {
-        // Read out the input name and convert it to a java.lang.String
-        char* inputName;
-        checkOrtStatus(jniEnv,api,api->SessionGetInputName((OrtSession*)sessionHandle, i, allocator, &inputName));
-        jstring name = (*jniEnv)->NewStringUTF(jniEnv,inputName);
-        (*jniEnv)->SetObjectArrayElement(jniEnv, array, i, name);
-        checkOrtStatus(jniEnv,api,api->AllocatorFree(allocator,inputName));
+JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_getInputNames(JNIEnv* jniEnv, jobject jobj, jlong apiHandle, jlong sessionHandle, jlong allocatorHandle) {
+  (void)jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  OrtAllocator* allocator = (OrtAllocator*)allocatorHandle;
+  OrtSession* session = (OrtSession*)sessionHandle;
+
+  // Setup
+  char* stringClassName = "java/lang/String";
+  jclass stringClazz = (*jniEnv)->FindClass(jniEnv, stringClassName);
+
+  // Get the number of inputs
+  size_t numInputs = 0;
+  OrtErrorCode code = checkOrtStatus(jniEnv, api, api->SessionGetInputCount(session, &numInputs));
+  if (code != ORT_OK) {
+    return NULL;
+  }
+
+  // Allocate the return array
+  jobjectArray array = (*jniEnv)->NewObjectArray(jniEnv, safecast_size_t_to_jsize(numInputs), stringClazz, NULL);
```

Comment:
>safecast_size_t_to_jsize [](http://example.com/codeflow?start=57&length=24)

I am still wondering how we could detect a problem and return an error here/

Reply:
You mean if numInputs overflowed a signed 32-bit integer? Is that not likely to cause problems elsewhere when you load one of those models in? It wouldn't fit in a protobuf.

Reply:
In this case, yes. But the helper is more general and we may apply it to data. Currently, it would call abort and Java would dump a text file. Would that text file have sufficient information to diagnose the issue? However, from the customer perspective, it would be better to see some kind of error message as they may not be aware of those dump files and report to you(us). Then we could solve it in seconds.

Reply:
Ok, I can refactor that later. It'll touch a bunch of files and it would be better to get this PR and the other one for OrtJniUtil finished before making that change.

Reply:
Our protobuf declarations actually allow unsigned datatypes

Reply:
I know, but protobuf has a message cap of 2GB, so you couldn't have 2^31 inputs in the same model without exceeding the file size.

Reply:
For inputs it is okay. Thinking more of the general usage.